sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsDelete.java

```diff
@@ -28,6 +28,25 @@
  */
 @Evolving
 public interface SupportsDelete {
+
+  /**
+   * Checks whether it is possible to delete data from a data source table that matches filter
+   * expressions.
+   * <p>
+   * Rows should be deleted from the data source iff all of the filter expressions match.
+   * That is, the expressions must be interpreted as a set of filters that are ANDed together.
+   * <p>
+   * Spark will call this method to check if the delete is possible without significant effort.
+   * Otherwise, Spark will try to rewrite the delete operation and produce row-level changes
+   * if the data source table supports deleting individual records.
+   *
+   * @param filters filter expressions, used to select rows to delete when all expressions match
+   * @return true if the delete operation can be performed
+   */
+  default boolean canDeleteWhere(Filter[] filters) {
```

Comment:
Shall we clarify the relation between this and `deleteWhere`? say `deleteWhere` is only invoked when `canDeleteWhere` is positive. I must admit I'm somewhat confused when looking at this two and not sure what assumption should I make.

Reply:
Let me update the doc to make sure it is clear.

Reply:
Done. Could you check if it is descriptive enough, @sunchao?

Reply:
Thanks for the update @aokolnychyi . I'm wondering if it makes sense to clearly point out that `canDeleteWhere` is called at planning time, say: "Spark will call this method **at planning time** to check if the delete is possible without significant effort."

Also for `deleteWhere`, I'm wondering if the following makes sense:
"Delete data from a data source table that matches filter expressions. **This is only invoked when the canDeleteWhere check succeed**".

Reply:
Makes sense to me, @sunchao. Updated.

Reply:
This was resolved.